partnerapp/src/InteractionDesignerEmbedView.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, useEffect } from "react";
+import { observer } from "mobx-react-lite";
+import { useParams, useHistory } from "react-router-dom";
+import store, { CampaignStore, Campaign } from "./campaign-store";
+
+interface InteractionDesignerEvent {
+  type: "interactiondesigner:cancel" | "interactiondesigner:save";
+  interaction_id: string;
+  organization_id: string;
+  user_id: string;
+}
+
+async function onInteractionDesignerEvent(
+  eventData: InteractionDesignerEvent,
+  store: CampaignStore,
+  campaign: Campaign,
+  history: { push: (arg0: string) => void }
+) {
+  console.log("postMessage Event handler received: ", eventData, campaign);
+  if (eventData.type === "interactiondesigner:cancel") {
+    await store.revertCampaign(campaign);
+    window.location.href = "/campaigns";
```

Comment:
This should probably use the history thing passed in?
```suggestion
    history.push("/campaigns");
```